music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,552 @@
+
+        for thisMeasure in self.container.getElementsByClass('Measure'):
+            # TimeSignatures
+            tsThisMeasure = thisMeasure.getElementsByClass('TimeSignature')
+            if tsThisMeasure:
+                firstTS = tsThisMeasure[0]
+                self.combinedList.append(f'Time Signature: {firstTS.ratioString}')
+                if len(tsThisMeasure) > 1:
+                    unprocessedTSs = [x.ratioString for x in tsThisMeasure[1:]]
+                    msg = f'further time signature change(s) unprocessed: {unprocessedTSs}'
+                    self.combinedList.append(f'Note: {msg}')
+
+            # RomanNumerals
+            measureString = ''  # Clear for each measure
+
+            rnsThisMeasure = thisMeasure.getElementsByClass('RomanNumeral')
+
+            for rn in rnsThisMeasure:
+                chordString = self.getChordString(rn)
+                measureString = rnString(measureNumber=thisMeasure.measureNumber,  # WithSuffix ?
+                                         beat=rn.beat,
+                                         chordString=chordString,
+                                         inString=measureString,  # Creating update
+                                         )
+
+            if measureString:
+                self.combinedList.append(measureString)
+
+    def getChordString(self,
+                       rn: roman.RomanNumeral):
+        '''
+        Produce a string from a Roman number with the chord and
+        the key if that key constitutes a change from the foregoing context.
+
+        >>> p = stream.Part()
+        >>> m = stream.Measure()
+        >>> m.insert(0, meter.TimeSignature('4/4'))
+        >>> m.insert(0, roman.RomanNumeral('V', 'G'))
+        >>> p.insert(0, m)
+        >>> testCase = romanText.writeRoman.RnWriter(p)
+        >>> sameKeyChord = testCase.getChordString(roman.RomanNumeral('I', 'G'))
+        >>> sameKeyChord
+        'I'
+
+        >>> changeKeyChord = testCase.getChordString(roman.RomanNumeral('V', 'D'))
+        >>> changeKeyChord
+        'D: V'
+        '''
+
+        keyString = rn.key.tonicPitchNameWithCase.replace('-', 'b')
+        if keyString != self.currentKeyString:
+            self.currentKeyString = keyString
+            return f'{keyString}: {rn.figure}'
+        else:
+            return str(rn.figure)
+
+
+# ------------------------------------------------------------------------------
+
+def rnString(measureNumber: int,
+             beat: Union[str, int, float, fractions.Fraction],
+             chordString: str,
+             inString: Optional[str] = ''):
+    '''
+    Creates or extends a string of RomanText such that the output corresponds to a single
+    measure line with one or more pairs of beat to Roman numeral.
+
+    If the inString is not given, None, or an empty string then this function starts a new line.
+
+    >>> lineStarter = romanText.writeRoman.rnString(14, 1, 'G: I')
+    >>> lineStarter
+    'm14 b1 G: I'
```

Comment:
looking at the spec, "b1 " is much better removed -- all chords at the beginning of the measure are "b1" chords.